### PR TITLE
Several refactorings

### PR DESCRIPTION
**Pull request by rachitnigam**
Not ready for review.

### CODE REVIEW

**Comment by rachitnigam** [comment]
Ready for review.

File: stopify-continuations/src/callcc/jumper.ts [hunk 1]
```diff
@@ -84,47 +83,34 @@ function func(path: NodePath<Labeled<FunctionT>>, state: State): void {
   }
   const restoreLocals = path.node.localVars;
 
-  // We instrument every non-flat function to begin with a *restore block*
```
Comment (arjunguha) [inline stopify-continuations/src/callcc/jumper.ts]: Why did you delete this comment? Bring it back.

File: stopify-continuations/src/callcc/jumper.ts [hunk 2]
```diff
@@ -84,47 +83,34 @@ function func(path: NodePath<Labeled<FunctionT>>, state: State): void {
...
-  // performance. So, we avoid restoring it faithfully unless we are explicitly
-  // configured to do so.
-  const restoreBlock: t.Statement[] = [ ];
-  // Restore all local variables. Creates the expression:
-  //     [local0, local1, ... ] = topStack.locals;
-  restoreBlock.push(
+  const frame = t.identifier('$frame');
+
+  const restoreBlock = t.blockStatement([
+    t.variableDeclaration('const', [t.variableDeclarator(frame, popRuntimeStack)]),
```
Comment (arjunguha) [inline stopify-continuations/src/callcc/jumper.ts]: I disagree with this refactoring. Undo it.

File: stopify-continuations/bin/build [hunk 3]
```diff
@@ -5,10 +5,16 @@ set -e
...
 mkdir -p dist/tmp
 
+# Generate pre-stopified versions of each runtime
+
+echo 'Generating pre-stopififed runtimes'
+
 for TRANSFORM in lazy eager retval fudge; do
   ./bin/stopify-continuations.js -t $TRANSFORM dist/src/runtime/implicitApps.js dist/tmp/implicitApps.$TRANSFORM.js
   ./bin/stopify-continuations.js -t $TRANSFORM dist/src/runtime/hofs.js dist/tmp/hofs.$TRANSFORM.js
-  ./bin/stopify-continuations.js -t $TRANSFORM dist/src/runtime/gettersSetters.js dist/tmp/gettersSetters.$TRANSFORM.js
```
Comment (arjunguha) [inline stopify-continuations/bin/build]: Why is this okay to do? What if a getter captures a stack?

File: stopify-continuations/src/callcc/jumper.ts (hunk 2 above)
Comment (rachitnigam) [reply stopify-continuations/src/callcc/jumper.ts]: Why do you disagree with this? It marginally faster and the output code is easier to read.

Comment (arjunguha) [reply stopify-continuations/src/callcc/jumper.ts]: I mean, you can change the output code. But, write the code in this style:

```
const restoreBlock: t.Statement[] = [];
restoreBlock.push(...);
````

File: stopify-continuations/bin/build (hunk 3 above)
Comment (rachitnigam) [reply stopify-continuations/bin/build]: Getters runtime has no function calls. All functions are trivially flat.

File: stopify-continuations/src/callcc/jumper.ts (hunk 2 above)
Comment (rachitnigam) [reply stopify-continuations/src/callcc/jumper.ts]: Oh, sure.

File: stopify-continuations/bin/build (hunk 3 above)
Comment (arjunguha) [reply stopify-continuations/bin/build]: You're right. I guess I'm confused how getter work at all.

File: stopify-continuations/src/callcc/jumper.ts (hunk 1 above)
Comment (rachitnigam) [reply stopify-continuations/src/callcc/jumper.ts]: Done.

File: stopify-continuations/src/callcc/jumper.ts (hunk 2 above)
Comment (rachitnigam) [reply stopify-continuations/src/callcc/jumper.ts]: Done.

File: stopify-continuations/src/callcc/jumper.ts [hunk 4]
```diff
@@ -87,20 +83,27 @@ function func(path: NodePath<Labeled<FunctionT>>, state: State): void {
...
-    t.expressionStatement(t.assignmentExpression('=',
-      t.arrayPattern(restoreLocals), t.memberExpression(topOfRuntimeStack,
-        t.identifier('locals')))));
+  const restoreBlock = [
+    t.variableDeclaration('const', [t.variableDeclarator(frame, popRuntimeStack)]),
+    t.expressionStatement(t.assignmentExpression('=', target,
+      t.memberExpression(frame, t.identifier('index')))),
+  ];
+
+  if (restoreLocals.length > 0) {
```
Comment (arjunguha) [inline stopify-continuations/src/callcc/jumper.ts]: How often do you find that `restoreLocals.length === 0`?

Comment (rachitnigam) [reply stopify-continuations/src/callcc/jumper.ts]: Anecdotally, I see it happen regularly when working with tests. 
I don't know exactly how often this happens in normal code.
